Sources/TuistKit/Generator/Generator.swift

```diff
@@ -243,9 +253,14 @@ class Generator: Generating {
 
         // Apply graph mappers
         let (updatedGraph, graphMapperSideEffects) = try graphMapperProvider.mapper(config: config).map(graph: graph)
-        let updatedWorkspace = updatedModels
+        var updatedWorkspace = updatedModels
             .workspace
-            .merging(projects: updatedGraph.projects.map(\.path))
+
+        updatedWorkspace.projects = Array(
+            Set(
+                updatedWorkspace.projects + updatedGraph.projects.map(\.path)
+            )
+        )
```

Comment:
minor: I believe `.merging(projects: updatedGraph.projects.map(\.path))` achieves the same result